mx.py

```diff
@@ -3590,6 +3590,8 @@ def update(self, vcdir, rev=None, mayPull=False, clean=False, abortOnError=False
         """
         if rev and mayPull and not self.exists(vcdir, rev):
             self.pull(vcdir, rev=rev, update=False, abortOnError=abortOnError)
+            if not self.exists(vcdir, rev):
+                abort('Fetch of %s succeeded\nbut did not contain requested revision %s.\nCheck that your remotes agree with the suite.py reposity location' % (vcdir, rev))
```

Comment:
Hm, yeah, I see your point regarding "git really wants you to do this through your configured remotes". We could probably attempt to detect the remote name by URL based on `git remote -v` output, but it's non-trivial, since it depends on the protocol used (https, ssh). So I'm fine with this.